caiman/source_extraction/volpy/utils.py

```diff
@@ -0,0 +1,281 @@
+    return Cn
+
+def quick_annotation(img, min_radius, max_radius, roughness=2):
+    """ Quick annotation method in VolPy using cell magic wand plugin
+    Args:
+        img: 2-D array
+            img as the background for selection
+            
+        min_radius: float
+            minimum radius of the selection
+            
+        max_radius: float
+            maximum raidus of the selection
+            
+        roughness: int
+            roughness of the selection surface
+            
+    Return:
+        ROIs: 3-D array
+            region of interests 
+            (# of components * # of pixels in x dim * # of pixels in y dim)
+    """
+    get_ipython().run_line_magic('matplotlib', 'auto')
+    def tellme(s):
+        print(s)
+        plt.title(s, fontsize=16)
+        plt.draw()
+        
+    keep_select=True
+    ROIs = []
+    while keep_select:
+        # Plot img
+        plt.clf()
+        plt.imshow(img, cmap='gray', vmax=np.percentile(img, 98))            
+        if len(ROIs) == 0:
+            pass
+        elif len(ROIs) == 1:
+            plt.imshow(ROIs[0], alpha=0.3, cmap='Oranges')
+        else:
+            plt.imshow(np.array(ROIs).sum(axis=0), alpha=0.3, cmap='Oranges')
+        
+        # Plot point and ROI
+        tellme('Click center of neuron')
+        center = plt.ginput(1)[0]
+        plt.plot(center[0], center[1], 'r+')
+        ROI = cell_magic_wand_single_point(img, (center[1], center[0]), 
+                                           min_radius=min_radius, max_radius=max_radius, 
+                                           roughness=roughness, zoom_factor=1)[0]
+        plt.imshow(ROI, alpha=0.3, cmap='Reds')
+    
+        # Select or not
+        tellme('Select? Key click for yes, mouse click for no')
+        select = plt.waitforbuttonpress()
+        if select:
+            ROIs.append(ROI)
+            tellme('You have selected a neuron. \n Keep selecting? Key click for yes, mouse click for no')
+        else:
+            tellme('You did not select a neuron \n Keep selecting? Key click for yes, mouse click for no')
+        keep_select = plt.waitforbuttonpress()
+        
+    plt.close()        
+    ROIs = np.array(ROIs)   
+    get_ipython().run_line_magic('matplotlib', 'inline')
+    return ROIs
+
+def mrcnn_inference(img, weights_path, display_result=True):
+    """ Mask R-CNN inference in VolPy
+    Args: 
+        img: 2-D array
+            summary images for detection
+            
+        weights_path: str
+            path for Mask R-CNN weight
+            
+        display_result: boolean
+            if True, the function will plot the result of inference
+        
+    Return:
+        ROIs: 3-D array
+            region of interests 
+            (# of components * # of pixels in x dim * # of pixels in y dim)
+    """
+    from caiman.source_extraction.volpy.mrcnn import visualize, neurons
+    import caiman.source_extraction.volpy.mrcnn.model as modellib
```

Comment:
Consider doing these imports at the head of the file rather than dynamically

Reply:
What I think is the following: if they use Mask R-CNN, then they will install keras to run these lines. If they don't use Mask R-CNN, they can skip these steps and no need to install keras.

Reply:
That's a fair point.